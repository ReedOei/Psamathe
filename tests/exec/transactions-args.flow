contract Main {
    transaction foo(n : ! nat) returns ! nat:
        var t : ! nat := 0
        n --> t
        12 --> t
        return t

    transaction add(n : ! nat, m : ! nat) returns ! nat:
        var t : ! nat := 0
        n --> t
        m --> t
        return t
}

;

var x : ! Main := new Main()
var a : ! nat := 0
var b : ! nat := 0

781 --> a
611 --> b

var res : ! nat := x.add(a,b)
if res != 1392 { revert("Failure!") }

