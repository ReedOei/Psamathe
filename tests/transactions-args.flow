contract Main {
    transaction foo(n : nat) returns nat:
        var t : nat
        n --> t
        12 --> t
        return t

    transaction add(n : nat, m : nat) returns nat:
        var t : nat
        n --> t
        m --> t
        return t
}

;

var x : Main
var a : nat
var b : nat

new Main() --> x
781 --> a
611 --> b

var res : nat
x.add(a,b)
x.add --> res

