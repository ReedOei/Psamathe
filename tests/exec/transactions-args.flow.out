<generatedTop>
  <common>
    <k>
      .
    </k>
    <contracts>
      SetItem ( Main )
    </contracts>
    <fields>
      Main |-> .Set
    </fields>
    <constructors>
      Main |-> constructor ( .VarDefs , pass )
    </constructors>
    <functions>
      Main |-> SetItem ( transaction ( add , var ( 1 ) : ! nat , n : ! nat , m : ! nat , .VarDefs , pass var t : ! nat := 0 in ( ( pass n --[ everything ]-> t ) ( pass m --[ everything ]-> t ) pass t --[ everything ]-> var ( 1 ) pack exit ) ) )
      SetItem ( transaction ( foo , var ( 0 ) : ! nat , n : ! nat , .VarDefs , pass var t : ! nat := 0 in ( ( pass n --[ everything ]-> t ) ( pass 12 --[ everything ]-> t ) pass t --[ everything ]-> var ( 0 ) pack exit ) ) )
    </functions>
    <types>
      Main |-> SetItem ( type ( Main , SetItem ( asset )
      SetItem ( consumable ) , Main ) )
      address |-> SetItem ( type ( address , SetItem ( consumable )
      SetItem ( nonfungible ) , address ) )
      bool |-> SetItem ( type ( bool , SetItem ( consumable )
      SetItem ( nonfungible ) , bool ) )
      nat |-> SetItem ( type ( nat , SetItem ( consumable )
      SetItem ( fungible ) , nat ) )
      string |-> SetItem ( type ( string , SetItem ( consumable )
      SetItem ( nonfungible ) , string ) )
    </types>
    <freshCounter>
      13
    </freshCounter>
  </common>
  <lookup>
    this |-> loc ( -1 )
  </lookup>
  <storage>
    loc ( 3 ) |-> resource ( Main , emptyval )
    loc ( 4 ) |-> resource ( nat , 781 )
    loc ( 5 ) |-> resource ( nat , 611 )
    loc ( 6 ) |-> resource ( nat , 0 )
    loc ( 7 ) |-> resource ( nat , 0 )
    loc ( 9 ) |-> resource ( nat , 0 )
    loc ( 10 ) |-> resource ( nat , 0 )
    loc ( 11 ) |-> resource ( nat , 0 )
    loc ( 12 ) |-> resource ( nat , 1392 )
    loc ( -1 ) |-> resource ( Base , .Map )
  </storage>
  <catchStack>
    .List
  </catchStack>
</generatedTop>
